internal/nix/input.go

```diff
@@ -50,46 +76,62 @@ func InputFromString(raw string, locker lock.Locker) *Input {
 	return &Input{*u, locker, raw, ""}
 }
 
+// InputFromProfileItem sets the raw Input as the `item`'s unlockedReference i.e.
+// the flake reference and output attribute path used at install time.
 func InputFromProfileItem(item *NixProfileListItem, locker lock.Locker) *Input {
 	return InputFromString(item.unlockedReference, locker)
 }
 
-func (i *Input) IsLocal() bool {
+// isLocal specifies whether this input is a local flake.
+// Usually, this is of the form: `path:./local_flake_subdir#myPackage`
+func (i *Input) isLocal() bool {
 	// Technically flakes allows omitting the scheme for local absolute paths, but
 	// we don't support that (yet).
 	return i.Scheme == "path"
 }
 
-func (i *Input) IsDevboxPackage() bool {
+// isDevboxPackage specifies whether this input is a nix package defined in a devbox.json config.
+// Usually, this is of the form: `name@version`.
+func (i *Input) isDevboxPackage() bool {
 	return i.Scheme == ""
 }
 
-func (i *Input) IsGithub() bool {
+// isGithub specifies whether this input is a remote flake hosted on a github repository.
+// example: github:nixos/nixpkgs/5233fd2ba76a3accb5aaa999c00509a11fd0793c#hello
+func (i *Input) isGithub() bool {
 	return i.Scheme == "github"
 }
 
 var inputNameRegex = regexp.MustCompile("[^a-zA-Z0-9-]+")
 
-func (i *Input) InputName() string {
+// FlakeInputName refers to the name of this Input that will be used in the
+// generated flake.nix. It is unique, and a slug is appended to avoid collisions.
+//
+// Note, that input name has nothing to do with the package name or flake name
+// that it may be referencing. That is the Input.raw field.
+func (i *Input) FlakeInputName() string {
 	result := ""
-	if i.IsLocal() {
+	if i.isLocal() {
 		result = filepath.Base(i.Path) + "-" + i.Hash()
-	} else if i.IsGithub() {
+	} else if i.isGithub() {
 		result = "gh-" + strings.Join(strings.Split(i.Opaque, "/"), "-")
-	} else if url := i.URLForInput(); IsGithubNixpkgsURL(url) {
-		u := HashFromNixPkgsURL(url)
-		if len(u) > 6 {
-			u = u[0:6]
+	} else if url := i.URLForFlake(); IsGithubNixpkgsURL(url) {
+		commitHash := CommitHashFromNixPkgsURL(url)
+		if len(commitHash) > 6 {
+			commitHash = commitHash[0:6]
 		}
-		result = "nixpkgs-" + u
+		result = "nixpkgs-" + commitHash
 	} else {
 		result = i.String() + "-" + i.Hash()
 	}
+
+	// replace all non-alphanumeric with dashes
 	return inputNameRegex.ReplaceAllString(result, "-")
 }
 
-func (i *Input) URLForInput() string {
-	if i.IsDevboxPackage() {
+// URLForFlake is the url to be used as the input in the generated flake.nix
+func (i *Input) URLForFlake() string {
+	if i.isDevboxPackage() {
```

Comment:
I'm not sure this is an improvement. This is literally the url that is used as an input in a flake.

Assume this struct is getting renamed to Package which will make the existing name less redundant

Reply:
The old name `URLForInput` conveys very little information, and is redundant. IMO the new name is much better.

Reply:
sounds good